Feat/training-datasets

Eysteinn-Orn opened the pull request:
added 1 folder, 2 files: "__init__.py" and "config.py" and a small class to __init__.py.
Config.py my be unnecessary at the current time.

njallskarp committed:
Merge pull request #2 from njallskarp/dev

Chore: Added config files for project setup

lsig committed:
Added validate function, need to make function evaluate model in another
script

Eysteinn-Orn committed:
Update __init__.py

lsig committed:
Refactored validate and trains so get_prediction and device live in
another file. Type interface for Answers also made

lsig committed:
Added path to files for ruff to test

lsig committed:
set maximum line to 200 for now

lsig committed:
Create lint.yml

lsig committed:
Changed return data type to tuple[torch.FloatTensor] from
torch.FloatTensor

njallskarp commented on training_datasets/__init__.py:
```diff
@@ -1,4 +1,18 @@
\ No newline at end of file
+
+
+class SquadDataset(torch.utils.data.Dataset):
```
Refactor this

njallskarp reviewed (changes requested):
Create a datasets.py module and put the Dataset class in there

lsig committed:
Update lint.yml

changed out old review dog version for a newer one

Eysteinn-Orn committed:
moved dataset to a new file and refactored

lsig committed:
Update lint.yml

Changed dependency strategy, see https://jacobian.org/til/github-actions-poetry/

Eysteinn-Orn committed:
organized inputs

lsig committed:
Changed poetry version

eysta00 commented:
organized imports*

lsig committed:
Removed second poetry install

njallskarp committed:
added metrics file

lsig commented:
I was wondering if you could format the docstrings so they are inline, otherwise looks good. Maybe I can setup a formatter.

njallskarp committed:
Merge pull request #10 from njallskarp/lsig-patch-1

Create lint.yml

njallskarp committed:
Merge pull request #11 from njallskarp/fix/config

Fix/config

njallskarp committed:
Merge pull request #13 from njallskarp/feature/metrics

added metrics file

njallskarp commented on training_datasets/datasets.py:
```diff
@@ -0,0 +1,27 @@
+            encodings (any): The preprocessed input encodings.
+            is_train (bool, optional): Whether the dataset is for training or not. Defaults to True.
+        """
+    def __init__(self, encodings: any, is_train: bool = True):
```
Add types, check in with @lsig if you need details on the types.

lsig replied:
Let me check

lsig replied:
It is the return type of tokenizer

lsig replied:
Probably Tokenzier().encodings or something like that

njallskarp reviewed (changes requested):
see comment. Need to improve typing

njallskarp commented on training_datasets/datasets.py:
```diff
@@ -0,0 +1,27 @@
+
+class SquadDataset(Dataset):
+    """
+    Code reused from open source colab notebook:
```
Class should not be nested, please review and fix

njallskarp reviewed (changes requested):
fix nesting of classes

eysta00 commented:
@lsig what is the type for encodings?

Eysteinn-Orn committed:
Update datasets.py

njallskarp committed:
fixed ruff errors in metrics

lsig committed:
Added docstring to Answers type

lsig committed:
Changed function name to get_device, much clearer

njallskarp committed:
Merge pull request #6 from njallskarp/feat/validate

Feat/validate, made validations function for model

njallskarp committed:
added main file

eysta00 commented:
> Collaborator
@lsig https://github.com/njallskarp/finetune-qa-powerset/pull/9#issuecomment-1609587840
are you talking about docstring comments beginning like this:
`""" Some text begins here

Args:
   The arguments

Returns:
    arguments added together`

lsig committed:
Extracting Metrics dict def to type declarations and renamed to Metrics

Eysteinn-Orn committed:
fixed docstring comment

njallskarp committed:
Merge pull request #19 from njallskarp/feature/adding-main-file

Feature/adding main file

njallskarp committed:
imported evaluate model

njallskarp committed:
Merge pull request #22 from njallskarp/bugfix/added-import-statement

bugfix/imported evaluate model

njallskarp committed:
adding codacity badge

njallskarp committed:
Merge pull request #24 from njallskarp/documentation/codacity-badge

Adding codacity badge

lsig committed:
Added tokenizers package to dependencies

njallskarp committed:
Merge pull request #21 from njallskarp/fix/metrics

Fix/metrics

njallskarp committed:
Update lint.yml

njallskarp committed:
Merge pull request #27 from njallskarp/njallskarp-patch-1

Update lint.yml

lsig committed:
Merge pull request #28 from njallskarp/main

# syncing dev <--> main

njallskarp committed:
Merge pull request #29 from njallskarp/fix/config

Added tokenizers package to dependencies

Eysteinn-Orn committed:
added datasets.py to training dataset

Eysteinn-Orn committed:
Merge branch 'revert-3-#chore(training-dataset-folders)' of https://github.com/njallskarp/finetune-qa-powerset into Feat/training-datasets
